src/menu.scss

```diff
@@ -5,165 +5,295 @@
-      outline-width: 1px;
+    .#{$block}__title {
+      @include fd-reset();
 
-      @include fd-var-color("outline-color", fd-color-state("hover", "action"), --fd-color-action-focus);
-    }
+      flex: 3 3 10%;  // needed to keep the icons aligned
+      color: $fd-menu-item-color;
 
-    @include fd-disabled() {
-      @include fd-var-color("outline-color", fd-color-state("disabled", "action"), --fd-color-action-disabled);
-      @include fd-var-color("color", $fd-forms-color--disabled, --fd-color-text-4);
+      @include fd-weight("normal");
 
-      cursor: not-allowed;
+      font-size: var(--sapFontSize);
+      font-family: var(--sapFontFamily);
       text-decoration: none;
+
+      @include fd-ellipsis();
     }
   }
 
-  &--addon-before {
-    .fd-menu__item {
-      padding-top: $fd-menu-item-padding-y;
-      padding-bottom: $fd-menu-item-padding-y;
-      padding-left: 2px;
-      padding-right: 2px;
+  &__addon-before {
+    @include fd-reset();
+    @include fd-flex-center();
+
+    // cozy tablet icon size
+    width: 2.25rem; // to account for item right padding of 16px or 1rem so 2.25rem-1rem = 1.25rem
+    height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+    margin: 0 0 0 -0.75rem;
+
+    @include fd-rtl() {
+      margin: 0 -1rem 0 0; // since before and after icons switch places, their margins also switch
     }
+
+    color: $fd-menu-primary-icon-color;
+
+    @include fd-active() {
+      color: $fd-menu-item-color--active;
+    }
+
+    @include fd-selected() {
+      color: $fd-menu-primary-icon-color;
+
+      @include fd-hover() {
+        color: $fd-menu-item-color--active;
+      }
+    }
+
+    display: flex;
+    background-color: transparent;
+
+    @include fd-icon-size("m", "before");
+    @include fd-icon-size("m", "after");
   }
 
-  &__addon-before {
+  // for affordance icon to be used with submenu
+  &__addon-after {
+    @include fd-reset();
     @include fd-flex-center();
 
+    flex: 0 0 auto;
+    display: flex;
+    color: $fd-menu-primary-icon-color;
+    // cozy tablet icon size
+    width: 2.25rem; // to account for item right padding of 16px or 1rem so 2.25rem-1rem = 1.25rem
+    height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+    margin: 0 -1rem 0 0;
+
     @include fd-rtl() {
-      float: right;
-      clear: right;
+      margin: 0 0 0 -0.75rem; // since before and after icons switch places, their margins also switch
+    }
+
+    @include fd-active() {
+      color: $fd-menu-item-color--active;
+    }
+
+    @include fd-selected() {
+      color: $fd-menu-primary-icon-color;
+
+      @include fd-hover() {
+        color: $fd-menu-item-color--active;
+      }
     }
 
     background-color: transparent;
-    float: left;
-    clear: left;
-    min-height: 1px;
-    width: 30px;
+
+    @include fd-icon-size("s", "before");
+    @include fd-icon-size("s", "after");
```

Comment:
This size should be applied only if the user wants affordance arrow. It shouldn't be applied for all addon-after. Maybe create a modifier class to handle the affordance arrow case

Reply:
so, would you say `fd-menu__addon-after` by default takes any icon in the normal font-size of 1rem but for affordance arrow I should keep a class like `fd-menu__addon-after--submenu` and handle the smaller size case here?